torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp

```diff
@@ -0,0 +1,211 @@
+#include <torch/csrc/jit/passes/onnx/scalar_type_analysis.h>
+#include <torch/csrc/jit/passes/dead_code_elimination.h>
+
+namespace torch {
+namespace jit {
+
+namespace onnx {
+using namespace ::c10::onnx;
+}
+
+namespace {
+class ScalarTypeHashFunction {
+ public:
+  size_t operator()(const c10::ScalarType& type) const {
+    return static_cast<size_t>(type);
+  }
+};
+
+static const std::unordered_map<c10::ScalarType, int, ScalarTypeHashFunction> scalarTypeToONNXTypeMap = {
+    {c10::kFloat, 1},
+    {c10::kByte, 2},
+    {c10::kChar, 3},
+    {c10::kShort, 5},
+    {c10::kInt, 6},
+    {c10::kLong, 7},
+    {c10::kBool, 9},
+    {c10::kHalf, 10},
+    {c10::kDouble, 11},
+};
+
+static int64_t ScalarTypeToONNXType(const c10::ScalarType& st) {
+  int64_t onnx_type = -1;
+  if (scalarTypeToONNXTypeMap.count(st) != 0) {
+    onnx_type = scalarTypeToONNXTypeMap.find(st)->second;
+  }
+  return onnx_type;
+}
+
+static const std::unordered_set<NodeKind> arithmeticOps = {
```

Comment:
I understand that the reason behind separating the ops in arithmeticOps and comparisonOps (below) is whether we can infer the output type based on inputs or not. It will be useful to spell out this reasoning as comments here.

Reply:
Thanks, added comments